Dynatrace/dynatrace-operator
# Add readOnlyRootFs check for OneAgent DaemonSet

## Description


Since v1.291 of the OneAgent the `securityContext. ReadOnlyRootFilesystem = true` is supported in the CloudNative and HostMonitoring cases.

This introduces a check that:
- if the version is older than  v1.291, we **don't** set `ReadOnlyRootFilesystem` to `true`
- if the the version is v1.291 or newer or unknown, we set `ReadOnlyRootFilesystem` to `true`

Related ticket: [K8S-5812](https://dt-rnd.atlassian.net/browse/)

## How can this be tested?


Deploy a older OneAgent version, see that the `ReadOnlyRootFilesystem` is `false`
Deploy a current OneAgent version, see that the `ReadOnlyRootFilesystem` is `true`
